### ISSUE TITLE
Help fixing documentation of association scopes

### ISSUE THREAD

**Issue by iamvery** [rails/rails#28165]
It appears the documentation for association scopes is misleading. However, it is not clear to me the optimal fix, so I'm opting to report the issue for feedback.

### Currently

At this time the [documentation for the `scope` option to a `has_many` association features the following example](https://github.com/rails/rails/blob/71152576cb80b7849895f48947689a750c07dffe/activerecord/lib/active_record/associations.rb#L1279):

```ruby
has_many :posts, ->(post) { where("max_post_length > ?", post.length) }
```

This implies that the value yielded to the scope block is one of the associated `post` records. This is true of the scope examples for each type of association, `has_one`, `belongs_to`, etc.

### Proposal

In my experience, this is not the behavior of the scope block on associations. Instead, they are yielded the instance of the current model whose association is being accessed. Therefore the example might be updated as such:

```ruby
has_many :posts, ->(author) { where("max_post_length > ?", author.average_post_length) }
```

Unfortunately updating the examples as such also seem to warrant wrapping them in the class definition, e.g. `class Author < ActiveRecord::Base`, which starts begging for more context. Not to mention this example itself doesn't make a ton of sense 😆.

### Conclusion

My first instinct was "no Jay, you must be wrong", because the documentation is _consistently_ incorrect for each type of association? So, am I wrong about this? (my tests confirm that I'm not wrong) Further, does anyone have suggestions about how to clarify the examples to an acceptable point?

I am happy to make the changes with some help in determining clarity, and as always thank you for such a wonderful tool in Rails and all your hard working in making it great! ❤️

**Comment by eugeneius** [comment]
The examples are consistently incorrect because they were all written by the same person around the same time: https://github.com/rails/rails/pulls?q=is%3Apr%20author%3Atomkadwill%20scope%20parameter

I think it should be possible to give examples with no surrounding context that still effectively demonstrate how to use the feature, if we pick the right models.

**Comment by iamvery** [comment]
I still intend to look at this.

**Comment by iamvery** [comment]
Looks like @rafaelfranca beat me to it. This was fixed by 0714251b2d0d422dc3ffc852eeb604281195b0c2
